src/soilbiogeochem/SoilBiogeochemDecompCascadeMIMICSMod.F90

```diff
@@ -0,0 +1,1024 @@
+         ! and supported by data in
+         ! Orchard, V.A., and F.J. Cook, 1983. Relationship between soil respiration
+         ! and soil moisture. Soil Biol. Biochem., 15(4):447-453.
+
+         do j = 1,nlevdecomp
+            do fc = 1,num_soilc
+               c = filter_soilc(fc)
+               psi = min(soilpsi(c,j),maxpsi)
+               ! decomp only if soilpsi is higher than minpsi
+               if (psi > minpsi) then
+                  w_scalar(c,j) = (log(minpsi/psi)/log(minpsi/maxpsi))
+               else
+                  w_scalar(c,j) = 0._r8
+               end if
+            end do
+         end do
+
+         if (use_lch4) then
+            ! Calculate ANOXIA
+            ! Check for anoxia w/o LCH4 now done in controlMod.
+
+            if (anoxia) then
+               do j = 1,nlevdecomp
+                  do fc = 1,num_soilc
+                     c = filter_soilc(fc)
+
+                     o_scalar(c,j) = max(o2stress_unsat(c,j), mino2lim)
+                  end do
+               end do
+            else
+               o_scalar(bounds%begc:bounds%endc,1:nlevdecomp) = 1._r8
+            end if
+         else
+            o_scalar(bounds%begc:bounds%endc,1:nlevdecomp) = 1._r8
+         end if
+
+      end if
+
+      ! Term that reduces decomposition rate at depth
+      do j = 1, nlevdecomp
+         do fc = 1, num_soilc
+            c = filter_soilc(fc)
+            if (use_vertsoilc) then
+               ! Using fixed e-folding depth as in
+               ! SoilBiogeochemDecompCascadeBGCMod.F90
+               depth_scalar(c,j) = exp(-zsoi(j) / decomp_depth_efolding)
+            else
+               depth_scalar(c,j) = 1.0_r8
+            end if
+         end do
+      end do
+
+      ! calculate rates for all litter and som pools
+      ! TODO Ok that I reversed order of do-loops for convenience?
+      do fc = 1,num_soilc
+         c = filter_soilc(fc)
+
+         ! TODO Move all to the params files
+         fmet_p1 = 0.85
+         fmet_p2 = 0.85
+         fmet_p3 = 0.013
+         tauMod_denom = 100.0_r8  ! For efficiency put _factor in params files
+         tauMod_factor = 1.0_r8 / tauMod_denom  ! ...instead of _denom?
+         tauMod_min = 0.8_r8
+         tauMod_max = 1.2_r8
+         tau_r1 = 5.2e-4_r8
+         tau_r2 = 0.3_r8  ! or 0.4?
+         tau_k1 = 2.4e-4_r8
+         tau_k2 = 0.1_r8
+
+         ! Time-dependent params from Wieder et al. 2015 & testbed code
+         ! TODO STILL MISSING N-related stuff: DIN...
+
+         ! TODO I don't see existing ligninNratioAvg, so calculate
+           ligninNratioAvg(c) =  &
+              (ligninNratio(c,leaf) * (cleaf2met(c) + cleaf2str(c)) + &
+               ligninNratio(c,froot) * (croot2met(c) + croot2str(c)) + &
+               ligninNratio(c,wood) * (cwd2str(c))) / &
+              max(0.001, cleaf2met(c) + cleaf2str(c) + &
+                         croot2met(c) + croot2str(c) + cwd2str(c))
```

Comment:
cleaf2met is leaf carbon to metabolic litter
similarly the others

rethink this as the sum of input to metabolic + str litter? @wwieder will look at this.

Reply:
What's the best way to do this for a column? 

We should be able to get lignin:N for each patch with data in `pftconMod.F90`:
- `lflitcn`, `frootcn` for litter CN as well as (are these used in CLM5?)
-  `lf_flig`, `fr_flig` for lignin (are these used for all pfts)?

We'd also likely need to keep track of the fluxes (`leafc_to_litter_patch`, `frootc__to_litter_patch`, `m_deadstemc_to_litter` from `CNVegCarbonFluxType.F90`) + associated N fluxes.  
Then all of these would have to be appropriately weighted from patches to columns.  Is there an easier way to go?

Reply:
Here's what I'm coming up with:

`lflitcn` is used in `CNPhenologyMod`

`frootcn` is used in `CNPhenologyMod`, `CNVegNitrogenStateType`, `NutrientCompetitionFlexibleCNMod`

`lf_flig` and `fr_flig` get assigned values for all pfts, though may not be used for all pfts

In `CNVegCarbonFluxType` I see that the relevant subroutine is `Summary_carbonflux` where we would use `call p2c` to get the variables to the column-level.

I see the following variable mappings:
`cleaf2met(c) + cleaf2str(c)`  -->  `leafc_to_litter_patch` after call p2c
`croot2met(c) + croot2str(c)`  -->  `frootc_to_litter_patch` after call p2c
`cwd2str(c)`  -->  `m_deadstemc_to_litter_patch` after call p2c
`ligninNratio(leaf)`  -->  `lf_flig * lflitcn`   is this how you would do this & the next one?
`ligninNratio(froot)`  -->  `fr_flig * frootcn`
`ligninNratio(wood)`  -->  `cwd_flig * cwdc_col(c) / cwdn_col(c)`

Reply:
I committed and pushed my first draft of this calculation.

Reply:
Thanks for all contributing